x/cronos/middleware/conversion_middleware.go

```diff
@@ -165,9 +165,10 @@ func (im IBCConversionModule) OnTimeoutPacket(
 		if err != nil {
 			return err
 		}
-		denom := im.getIbcDenomFromPacketAndData(packet, data)
-		if im.canBeConverted(ctx, denom) {
-			err = im.convertVouchers(ctx, data, denom, true)
+		if transferTypes.ReceiverChainIsSource(packet.GetSourcePort(), packet.GetSourceChannel(), data.Denom) {
```

Comment:
why do you skip when receiver chain is not source?

Reply:
I only revert to [previous](https://github.com/thomas-nguy/cronos/blob/51c7583b02eb69eaab28a4f2ff83c016d9d047c0/x/cronos/middleware/conversion_middleware.go#L174) timeout logic

Reply:
I have added the integration tests for bidirectional flow (non source tokens)

https://github.com/crypto-org-chain/cronos/pull/610

you can use this to verify the logic

Reply:
you mean for timeout case? but current main refund on denom `basecro` with path `""` instead of previous denom `ibc/6411AE2ADA1E73DB59DB151A8988F9B7D5E7E233D8414DB6817F8F1A01611F86` with path `transfer/channel-0`, without revert, refund [test](https://github.com/crypto-org-chain/cronos/blob/6056919f23b1e7f4a572fc649869dbfd3bb83a70/integration_tests/test_ibc_timeout.py#L81) will fail

Reply:
Yeah what needs to be changed is the `getIbcDenomFromPacketAndData` function.

The test should also work for non source token.

I think we need to introduce a boolean to this function and reverse the logic. Because in case of refund, the destination chain is actually the source chain